### ISSUE TITLE
SASL for version 3.1 does not include why the server sends 908 numeric

### ISSUE THREAD

**Issue by ChickenNuggers** [ircv3/ircv3-specifications#254]
```
22:10:44 <Andrio> In 'IRCv3.1 SASL Authentication' it's mentioned that the client can request a list of authentication mechanisms supported by the server.
22:10:46 <Andrio> How does that work?
22:18:21 <LordRyan> That is actually not documented.
```

**Comment by grawity** [comment]
3.1 never actually had that. 908 is just sent on reply to any AUTHENTICATE that requests an unsupported mechanism, so _technically_ a client could start with e.g. `*` or another garbage mechanism that would result in failure+mechlist 100% of the time. But that never was part of the spec.

**Comment by ChickenNuggers** [comment]
So it's not required; should something be added to show the list of supported mechanisms?

**Comment by grawity** [comment]
(Also not all servers _do_ send 908 at all – e.g. ircd-seven doesn't support it – so it can't be relied on. If you plan on implementing automatic selection: if the mechanism you just tried was immediately refused after 0 steps, remove it from your list, and if you _happen to_ receive a 908, use it to remove unsupported mechanisms from your list.)

**Comment by grawity** [comment]
I guess 3.1 could use an errata making 908 required, but 3.2's CAP & sasl specs have already added a better method anyway (mechlist in the CAP LS response directly).

**Comment by dequis** [comment]
-1 for errata, let's keep sasl-3.1 as is, documenting existing implementations. Clients should know how to handle 908 but nothing requires implementations to send it.

**Comment by attilamolnar** [comment]
-1 for errata requiring 908, possible fix https://github.com/ircv3/ircv3-specifications/compare/master...attilamolnar:sasl-908

**Comment by grawity** [comment]
Looks okay, but I'd still go with "SHOULD" over "MAY".

**Comment by dequis** [comment]
That works, seems more accurate than the existing version. I didn't realize the main issue here isn't the fact that it's vague about 908, but the fact that it references a method that doesn't exist to get the 908. Maybe mention it like that in the errata?

I'm indifferent about SHOULD vs MAY, it's not like this wording will get existing sasl-3.1 server implementations fixed, it's all about 3.2 at this point.

**Comment by attilamolnar** [comment]
> Looks okay, but I'd still go with "SHOULD" over "MAY".

I used MAY because this is a retroactive addition, but I'm indifferent as well.
